src/megachatapi_impl.cpp

```diff
@@ -9933,6 +10044,98 @@ MegaChatGeolocation *JSonUtils::parseGeolocation(rapidjson::Document &document)
+    }
+
+    auto mp4srcIterator = document.FindMember("src");
+    string mp4srcString;
+    if (mp4srcIterator != document.MemberEnd() && mp4srcIterator->value.IsString())
+    {
+        mp4srcString.assign(mp4srcIterator->value.GetString(), mp4srcIterator->value.GetStringLength());
+    }
+    else
+    {
+        API_LOG_ERROR("parseGiphy: invalid JSON struct - \"src\" field not found");
+        return std::unique_ptr<MegaChatGiphy>(nullptr);
+    }
+
+    auto webpIterator = document.FindMember("src_webp");
+    string webpsrcString;
+    if (webpIterator != document.MemberEnd() && webpIterator->value.IsString())
+    {
+        webpsrcString.assign(webpIterator->value.GetString(), webpIterator->value.GetStringLength());
+    }
+    else
+    {
+        API_LOG_ERROR("parseGiphy: invalid JSON struct - \"src_webp\" field not found");
+        return std::unique_ptr<MegaChatGiphy>(nullptr);
+    }
+
+    auto mp4sizeIterator = document.FindMember("s");
+    long mp4Size = 0;
+    if (mp4sizeIterator != document.MemberEnd() && mp4sizeIterator->value.IsString())
+    {
+        auto mp4sizeString = mp4sizeIterator->value.GetString();
+        mp4Size = atol(mp4sizeString);
+    }
+    else
+    {
+        API_LOG_ERROR("parseGiphy: invalid JSON struct - \"s\" field not found");
+        return std::unique_ptr<MegaChatGiphy>(nullptr);
+    }
+    
+    auto webpsizeIterator = document.FindMember("s_webp");
+    long webpSize = 0;
+    if (webpsizeIterator != document.MemberEnd() && webpsizeIterator->value.IsString())
+    {
+        auto webpsizeString = webpsizeIterator->value.GetString();
+        webpSize = atol(webpsizeString);
+    }
+    else
+    {
+        API_LOG_ERROR("parseGiphy: invalid JSON struct - \"s_webp\" field not found");
+        return std::unique_ptr<MegaChatGiphy>(nullptr);
+    }
+
+    auto giphywidthIterator = document.FindMember("w");
+    int giphyWidth = 0;
+    if (giphywidthIterator != document.MemberEnd() && giphywidthIterator->value.IsString())
+    {
+        auto giphywidthString = giphywidthIterator->value.GetString();
+        giphyWidth = atoi(giphywidthString);
+    }
+    else
+    {
+        API_LOG_ERROR("parseGiphy: invalid JSON struct - \"w\" field not found");
+        return std::unique_ptr<MegaChatGiphy>(nullptr);
+    }
+
+    auto giphyheightIterator = document.FindMember("h");
+    int giphyHeight = 0;
+    if (giphyheightIterator != document.MemberEnd() && giphyheightIterator->value.IsString())
+    {
+        auto giphyheightString = giphyheightIterator->value.GetString();
+        giphyHeight = atoi(giphyheightString);
+    }
+    else
+    {
+        API_LOG_ERROR("parseGiphy: invalid JSON struct - \"h\" field not found");
+        return std::unique_ptr<MegaChatGiphy>(nullptr);
+    }
+    return std::unique_ptr<MegaChatGiphyPrivate> (new MegaChatGiphyPrivate(mp4srcString, webpsrcString, mp4Size, webpSize, giphyWidth, giphyHeight, giphyTitle));
```

Comment:
It's C++11. So, can't use make_unique yet.

Reply:
You can use `::mega::make_unique<>` in the meantime :smiley: